packages/runtime/container-runtime/src/dataStoreContext.ts

```diff
@@ -375,12 +390,26 @@ export abstract class FluidDataStoreContext extends EventEmitter implements
         // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
         const entries = await this.channel!.snapshotInternal(fullTree);
 
-        const attributesBlob = createAttributesBlob(pkg);
+        const attributesBlob = createAttributesBlob(pkg, this.isRootStore, this.referenced);
         entries.push(attributesBlob);
 
         return { entries, id: null };
     }
 
+    public getReferencedRoutes(): string[] {
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        return this.channel!.getReferencedRoutes();
+    }
+
+    public collectUnreferencedFluidObjects(referencedObjectRoutes: string[]) {
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        this.channel!.collectUnreferencedFluidObjects(referencedObjectRoutes);
```

Comment:
If data store was not loaded (i.e. prior results were used), I assume we will not try to load it to clean it, right?

Reply:
Yes, we should never load anything unnecessarily.